module/os/freebsd/zfs/zfs_dir.c

```diff
@@ -368,18 +370,19 @@ static int
 zfs_purgedir(znode_t *dzp)
 {
 	zap_cursor_t	zc;
-	zap_attribute_t	zap;
+	zap_attribute_t	*zap;
 	znode_t		*xzp;
 	dmu_tx_t	*tx;
 	zfsvfs_t	*zfsvfs = dzp->z_zfsvfs;
 	int skipped = 0;
 	int error;
 
+	zap = zap_attribute_alloc();
```

Comment:
I see this is used only for extended attributes directory, so IF they are really limited to 255 chars, this may be fine.  Though can't say that I like IF's like that.

Reply:
fixed

Reply:
Yeah, I revert it back. I forgot it's only for xattr like you said. So it's not needed.